### ISSUE TITLE
Slider goes back when moved

### ISSUE THREAD

**Issue by pietro-fragnito** [angular-slider/angularjs-slider#346]
### Steps to reproduce
1) Install slider
2) Include slider
3) Move slider

### Expected behavior
Slider doesn't go back when moved.

### Actual behavior 
The slider shows a strange behavior: when I start to move the slider (also for a very very little step ahead or behind) it goes always back of a big step. I know it's difficult to understand so I loaded this [video](https://youtu.be/MON_IlwEZcE) on YouTube. And this is my slider (HTML + Controller):

`<rzslider
            rz-slider-model="slider.min"
            rz-slider-high="slider.max"
            rz-slider-options="slider.options">
    </rzslider>`

`$scope.slider = 
            min: 100,
            max: 180,
            options: {
                floor: 0,
                ceil: 200,
                step: 5,
                noSwitching: true,
                getSelectionBarColor: function () {
                    return '#F06820';
                },
                getPointerColor: function () {
                    return '#042863';
                },
                translate: function (value) {
                    return value + ' kW';
                },
                onEnd: function() { 
                     //DO SOMETHING
                }
            }
        };`

**Comment by ValentinH** [comment]
What version are you using / what browser?

I just set up a demo using your code and nothing seems broken: http://jsfiddle.net/yjeamcpu/

**Comment by pietro-fragnito** [comment]
I tried with Safari and Chrome: same behavior

**Comment by pietro-fragnito** [comment]
With Firefox it runs well

**Comment by ValentinH** [comment]
Do you have the issue on the demo I provided?

**Comment by pietro-fragnito** [comment]
Do you mean to create a new project like your demo?

**Comment by ValentinH** [comment]
Nope, I set up a demo and would like to know if you can reproduce your issue on it? If not, then please provide a similar demo showing the issue.

**Comment by pietro-fragnito** [comment]
I think I'm using an old version then your demo :

`{
  "name": "angularjs-slider",
  "version": "2.11.0",
  "description": "AngularJS slider directive with no external dependencies. Mobile friendly!.",
  "main": "dist/rzslider.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/angular-slider/angularjs-slider"
  },
  "keywords": [
    "angular",
    "slider"
  ],
  "scripts": {
    "commit": "git-cz",
    "build": "grunt",
    "test": "grunt test",
    "report-coverage": "cat ./tests/coverage/lcov.info | codecov"
  },
  "peerDependencies": {
    "angular": "^1.2.x"
  },
  "devDependencies": {
    "angular": "1.4.7",
    "angular-mocks": "1.4.8",
    "chai": "^3.4.1",
    "codecov.io": "^0.1.6",
    "commitizen": "^2.4.6",
    "cz-conventional-changelog": "^1.1.5",
    "grunt": "~0.4.2",
    "grunt-angular-templates": "^0.5.7",
    "grunt-contrib-concat": "^0.5.1",
    "grunt-contrib-mincss": "~0.3.2",
    "grunt-contrib-uglify": "~0.2.2",
    "grunt-contrib-watch": "^0.6.1",
    "grunt-karma": "^0.12.1",
    "grunt-ng-annotate": "^1.0.1",
    "grunt-recess": "~0.4.0",
    "grunt-replace": "^0.11.0",
    "grunt-serve": "^0.1.6",
    "karma": "^0.13.15",
    "karma-chai": "^0.1.0",
    "karma-chrome-launcher": "^0.2.2",
    "karma-coverage": "^0.5.3",
    "karma-mocha": "^0.2.1",
    "karma-ng-html2js-preprocessor": "^0.2.0",
    "karma-phantomjs-launcher": "^0.2.1",
    "karma-sinon": "^1.0.4",
    "mocha": "^2.3.4",
    "phantomjs": "^1.9.19",
    "recess": "~1.1.9",
    "sinon": "^1.17.2"
  },
  "author": "Rafal Zajac <rzajac@gmail.com>, Valentin Hervieu <valentin@hervieu.me>, Jussi Saarivirta <jusasi@gmail.com>, Angelin Sirbu <angelin.sirbu@gmail.com>",
  "license": "MIT",
  "readmeFilename": "README.md",
  "czConfig": {
    "path": "node_modules/cz-conventional-changelog"
  }
}`

How can I update via bower? I tried with 

`bower update angularjs-slider`

but it updated to the 2.11.0

**Comment by ValentinH** [comment]
It's because the latest version is 4.0.0 which is a major update version so you need to change your bower  file to explicitly require the 4.0.

**Comment by pietro-fragnito** [comment]
Ok. Updated to 4.0.2 but problem persists. I try to add a new slider to my project with your demo code, but I got same result.  It's difficult to reproduce it cause it's a very big app...in fact I make a demo video.

:(

**Comment by ValentinH** [comment]
You need to call this code:
```js
$timeout(function () {
        $scope.$broadcast('rzSliderForceRender');
    });
```
when your slider is  rendered. It can be at the initialisation of your controller or directive, depends on your architecture.

**Comment by pietro-fragnito** [comment]
I put it on "onEnd" 

```
// Slider
        $scope.slider = {
            min: 100,
            max: 180,
options: {
                floor: 0,
                ceil: 200,
                step: 5,
                noSwitching: true,
                getSelectionBarColor: function () {
                    return '#F06820';
                },
                getPointerColor: function () {
                    return '#042863';
                },
                translate: function (value) {
                    return value + ' kW';
                },
                onEnd: function () {

                    WidgetService.M_02_02_poiRange.query({
                        aggregator_id: "aggr_1",
                        min: $scope.slider.min,
                        max: $scope.slider.max
                    }, function (profiles) {
                        $scope.points.coords = profiles.coords;
                        $scope.polylines.coords = [profiles.coords];
                        $scope.map.center = profiles.center;
                        $scope.points.properties = profiles.properties;
                    }).$promise;

                    refreshSlider();
                }
            }
        };

        var refreshSlider = function () {
            $timeout(function () {
                $scope.$broadcast('rzSliderForceRender');
            });
        };
```

but the problem persists

**Comment by ValentinH** [comment]
No this is wrong. You should call it at the root of your component. At the same moment you declare your $scope.slider =

**Comment by pietro-fragnito** [comment]
Can you integrate this thing in your demo please?

**Comment by ValentinH** [comment]
http://jsfiddle.net/f83b002z/

**Comment by pietro-fragnito** [comment]
I got the cause of this behavior: in my CSS I set zoom property:

`body {
    -moz-transform: scale(0.9, 0.9); /* Moz-browsers */
    zoom: 0.85; /* Other non-webkit browsers */
    zoom: 85%; /* Webkit browsers */
    background-image: url("../img/bg.jpg");
    background-size: cover;
}`

if I remove this property the slider works well. Also if I set the zoom poverty of the child element where is the slider to 120% it works well.

Hope this can help.

**Comment by ValentinH** [comment]
If your slider is scaled, you can tell it to the library by setting the scale option so that all events work properly.

**Comment by pietro-fragnito** [comment]
Which value should I set?

**Comment by ValentinH** [comment]
You should set it to 1/your_css_scale: http://jsfiddle.net/tzjzt1ta/

**Comment by pietro-fragnito** [comment]
Sorry but the jsfiddle DEMO reports exactly the same strange behavior I describe.

:)

**Comment by ValentinH** [comment]
I don't see any weird behavior in the demo... the slider is scaled but works.

**Comment by pietro-fragnito** [comment]
Check this:

https://youtu.be/ZeeW-s6F2F4

Anyway my value should be 

`scale: 1/0.85`

but this doesn't work anyway

**Comment by ValentinH** [comment]
What browser are you using?

**Comment by pietro-fragnito** [comment]
Chrome (same behavior on Safari)

**Comment by ValentinH** [comment]
This is what I see: 

![](http://g.recordit.co/yNXqTQJFQF.gif)

**Comment by pietro-fragnito** [comment]
Yes, and if you look well, at the firs little move, the slider goes a little bit back and works well going on and back normally.

I solve it setting 

`zoom: 1.2; /* Other non-webkit browsers */
    zoom: 120%; /* Webkit browsers */`

in the CSS of the element where the slider is

**Comment by ValentinH** [comment]
I understand the bug you mention. Actually it is a bug in the scale option logic: it doesn't take the scale option into account for the slider handle size...

**Comment by ValentinH** [comment]
I'll fix it later

**Comment by pietro-fragnito** [comment]
Great! Thanks! In the meanwhile people have a workaround. 

:)

**Comment by ValentinH** [comment]
Actually, I don't see how your work-around solves the issue. Could you please update my demo to show me your work-around?

**Comment by pietro-fragnito** [comment]
Just done. The workaround is more effective in slider with ticks.

**Comment by ValentinH** [comment]
Could you share the link? :D

**Comment by pietro-fragnito** [comment]
I update the same you provided before http://jsfiddle.net/tzjzt1ta/

**Comment by ValentinH** [comment]
Yes but you need to fork it so I can see the changes..

**Comment by pietro-fragnito** [comment]
http://jsfiddle.net/piefragn/hwpdtzn4/

**Comment by ValentinH** [comment]
OK, this is what I thought: now your slider is not scaled anymore. The goal of the `scale` option is to keep it scaled but with working events.

**Comment by ValentinH** [comment]
Fixed and released in 5.8.0
